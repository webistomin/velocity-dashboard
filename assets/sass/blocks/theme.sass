html.theme
  +m('shelob')
    --color-primary: var(--color-blue)
    --color-secondary: var(--color-lavender)
    --color-tertiary: var(--color-gray-3)

  +m('denethor')
    --color-primary: var(--color-purple)
    --color-secondary: var(--color-darkturquoise)
    --color-tertiary: var(--color-yellow-lighter)

  +m('quickbeam')
    --color-primary: var(--color-deepskyblue)
    --color-secondary: var(--color-royalblue)
    --color-tertiary: var(--color-darkgreen)

  +m('shadowfax')
    --color-primary: var(--color-midnightblue)
    --color-secondary: var(--color-lawngreen)
    --color-tertiary: var(--color-dimgray)

  +m('grima')
    --color-primary: var(--color-darkorchid)
    --color-secondary: var(--color-pink)
    --color-tertiary: var(--color-lightslategray)
