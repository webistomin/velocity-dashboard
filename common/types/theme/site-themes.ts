export enum SiteThemes {
  SHELOB = 'shelob',
  DENETHOR = 'denethor',
  QUICKBEAM = 'quickbeam',
  SHADOWFAX = 'shadowfax',
  GRIMA = 'grima',
}
