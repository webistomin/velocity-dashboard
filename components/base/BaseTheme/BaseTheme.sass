+b('base-theme')

  +e('label')
    display: flex
    flex-direction: column

  +e('theme')
    display: block
    width: 100%
    max-width: rem(140px)
    height: rem(65px)
    margin-bottom: rem(14px)
    border-radius: 5px

    +m('shelob')
      background-image: linear-gradient(to right, rgb(var(--color-blue)) 33%, rgba(var(--color-blue), 0.2) 33%, rgba(var(--color-blue), 0.2) 66%,  rgb(var(--color-gray-3)) 33%)

    +m('denethor')
      background-image: linear-gradient(to right, rgb(var(--color-purple)) 33%, rgb(var(--color-darkturquoise)) 33%, rgb(var(--color-darkturquoise)) 66%, rgb(var(--color-yellow-lighter)) 33%)

    +m('quickbeam')
      background-image: linear-gradient(to right, rgb(var(--color-deepskyblue)) 33%, rgb(var(--color-royalblue)) 33%, rgb(var(--color-royalblue)) 66%, rgb(var(--color-darkgreen)) 33%)

    +m('shadowfax')
      background-image: linear-gradient(to right, rgb(var(--color-midnightblue)) 33%, rgb(var(--color-lawngreen)) 33%, rgb(var(--color-lawngreen)) 66%, rgb(var(--color-dimgray)) 33%)

    +m('grima')
      background-image: linear-gradient(to right, rgb(var(--color-darkorchid)) 33%, rgb(var(--color-pink)) 33%, rgb(var(--color-pink)) 66%, rgb(var(--color-lightslategray)) 33%)

  +e('text')
    font: var(--text-button)
    position: relative
    padding-left: rem(24px)

    &::before
      content: ''
      box-sizing: border-box
      width: rem(16px)
      height: rem(16px)
      +vertical-center()
      left: 0
      border: rem(1px) solid rgb(var(--color-gray-3))
      border-radius: 50%
      transition: border 0.2s ease-out

  +e('radio')
    &:checked ~ .base-theme__label
      .base-theme__text::before
        border: rem(4px) solid rgb(var(--color-primary))

body.using-keyboard
  .base-theme:focus-within
    outline: rem(1px) solid rgb(var(--color-primary))

  .base-theme__radio:focus ~ .base-radio__label
    outline: rem(1px) solid rgb(var(--color-primary))
