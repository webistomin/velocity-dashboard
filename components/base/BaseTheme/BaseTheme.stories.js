import { text, boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import BaseTheme from './BaseTheme';

export default {
  title: 'BaseTheme',
  parameters: {
    options: {
      showPanel: true,
      selectedPanel: 'storybook/knobs/panel',
    },
  },
};

export const Default = () => ({
  components: { BaseTheme },
  props: {
    label: {
      type: String,
      default: text('label', 'Text'),
      required: true,
    },
    id: {
      type: String,
      default: text('id', 'base-form-group'),
      required: true,
    },
    text: {
      type: String,
      default: text('text', ''),
      required: true,
    },
    theme: {
      type: String,
      default: text('theme', ''),
      required: true,
    },
    checked: {
      type: Boolean,
      default: boolean('checked', false),
    },
  },
  methods: {
    onInput: action('emit input'),
  },
  data() {
    return {
      currentTheme: '',
    };
  },
  template: `<div style="padding: 40px;">
    <BaseTheme
      :id="'setting-theme-shelob'"
      :theme="'shelob'"
      :value="'shelob'"
      :label="'shelob'"
      :checked="this.currentTheme === 'shelob'"
      @input="onInput(), currentTheme = 'shelob'"
    />
    <BaseTheme
      :id="'setting-theme-denethor'"
      :theme="'denethor'"
      :value="'denethor'"
      :label="'denethor'"
      :checked="this.currentTheme === 'denethor'"
      @input="onInput(), currentTheme = 'denethor'"
    />
  </div>`,
});
