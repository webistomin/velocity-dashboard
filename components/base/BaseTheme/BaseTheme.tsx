import { VueComponent } from 'types/vue-components';
import { Component, Prop } from 'nuxt-property-decorator';
import { VNode } from 'vue';

import './BaseTheme.sass';
import { Emit } from '~/node_modules/nuxt-property-decorator';

export type IBaseThemes = 'shelob' | 'denethor' | 'quickbeam' | 'shadowfax' | 'grima';

export interface IBaseThemeProps {
  value: string;
  id: string;
  label: string;
  theme: IBaseThemes;
  checked?: boolean;
  onInput?: (event: Event) => void;
}

@Component({
  name: 'BaseTheme',
})
export default class BaseTheme extends VueComponent<IBaseThemeProps> {
  @Prop()
  private readonly value!: IBaseThemeProps['value'];

  @Prop()
  private readonly id!: IBaseThemeProps['id'];

  @Prop()
  private readonly label!: IBaseThemeProps['label'];

  @Prop()
  private readonly theme!: IBaseThemeProps['theme'];

  @Prop()
  private readonly checked!: IBaseThemeProps['checked'];

  @Emit('input')
  onInput(event: Event): Event {
    return event;
  }

  public render(): VNode {
    return (
      <div class='base-theme'>
        <input
          class='base-theme__radio visually-hidden'
          type='radio'
          name='theme-selection'
          id={this.id}
          value={this.value}
          onInput={this.onInput}
          checked={this.checked}
        />
        <label for={this.id} class='base-theme__label label'>
          <span class={`base-theme__theme base-theme__theme_${this.theme}`} />
          <span class='base-theme__text'>{this.label}</span>
        </label>
      </div>
    );
  }
}
