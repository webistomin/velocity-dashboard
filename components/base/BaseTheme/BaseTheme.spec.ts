import { mount } from '@vue/test-utils';
import { SiteThemes } from 'common/types/theme/site-themes';
import BaseTheme from './BaseTheme';

describe('BaseTheme', () => {
  const themeSelector = '.base-theme__radio';

  const factory = (propsData = {}, options = {}) => {
    return mount(BaseTheme, {
      propsData: {
        theme: SiteThemes.SHELOB,
        value: '',
        id: 'id',
        label: 'label',
        ...propsData,
      },
      stubs: ['VSelect'],
      ...options,
    });
  };

  it('Matches snapshot', () => {
    const wrapper = factory().html();
    expect(wrapper).toMatchSnapshot();
  });

  it('Emit "input" event', () => {
    const wrapper = factory();
    const input = wrapper.find(themeSelector);
    input.trigger('input');
    expect(wrapper.emitted().input);
  });
});
