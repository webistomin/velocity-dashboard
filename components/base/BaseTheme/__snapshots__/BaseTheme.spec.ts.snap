// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTheme Matches snapshot 1`] = `"<div class=\\"base-theme\\"><input type=\\"radio\\" name=\\"theme-selection\\" id=\\"id\\" class=\\"base-theme__radio visually-hidden\\" value=\\"\\"><label for=\\"id\\" class=\\"base-theme__label label\\"><span class=\\"base-theme__theme base-theme__theme_shelob\\"></span><span class=\\"base-theme__text\\">label</span></label></div>"`;
